sys/newlib/thread_safe/Makefile.include

```diff
@@ -0,0 +1,22 @@
+# define the compile test file
+define __has_reent_small_test
+#include <sys/reent.h>\n
+#ifndef _REENT_SMALL\n
+#error wasting 1KB\n
+#endif
+endef
+
+# compile the tests
+__has_reent := $(shell echo "\#include <sys/reent.h>\n" | $(CC) -E $(CFLAGS) $(INCLUDES) - 2> /dev/null > /dev/null; echo $$?)
+__has_reent_small := $(shell echo "$(__has_reent_small_test)" | $(CC) -E $(CFLAGS) $(INCLUDES) - 2> /dev/null > /dev/null; echo $$?)
+
+# check if we use REENT_SMALL
+ifeq (0, $(__has_reent))
+  ifeq (0, $(__has_reent_small))
+    export USE_NEWLIB_THREAD_SAFE := 1
```

Comment:
the export doesn't generate an environment variable.
This pull request works when I change it to `CFLAGS += -DUSE_NEWLIB_THREAD_SAFE`